jasmine/src/string-extension.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+String.prototype.hasVowels = function() {
+  //Returns true if the string contains vowels.
+
+  return /[aeiou]/i.test(this);
+};
+
+String.prototype.toUpper = function() {
+  //Returns the String in question but with all characters in upper cases as applicable.
+
+  return this.replace(/[a-z]/g, function(letter) {
+    return String.fromCharCode(letter.charCodeAt(0) & ~32);
+  });
+};
+
+String.prototype.toLower = function() {
+  //Returns the String in question but with all characters in their lower cases as applicable.
+
+  return this.replace(/[A-Z]/g, function(letter) {
+    return String.fromCharCode(letter.charCodeAt(0) | 32);
+  });
```

Comment:
Use of bitwise operators makes the code obscure. Addition would achieve the same result while making the code more self-documenting.

Same case applies to [line 12](https://github.com/andela-ekahindi/string-extension/blob/f747ff7173e3a8ace3d1e807ccaa8826430288fb/jasmine/src/string-extension.js#L12) with substraction.